installers/osie/mirror.go

```diff
@@ -42,6 +43,22 @@ func buildOSIEURL() (*url.URL, error) {
 	return u, nil
 }
 
+func getOSIEPathOverride() string {
+	base, err := url.Parse(conf.MirrorBaseUrl)
+	if err != nil {
+		panic(errors.Wrap(err, "parsing MirrorBaseUrl"))
+	}
+	if s, ok := os.LookupEnv("OSIE_PATH_OVERRIDE"); ok {
```

Comment:
Hey @ptrivedi. I know looking at this file the model is to use env vars here but it's something that should be avoided. Boots' design philosophy is that 
>_"Only func main has the right to decide which flags, env variables, config files are available to the user"_ -- [ref](https://github.com/tinkerbell/boots/blob/main/docs/DESIGNPHILOSOPHY.md#functionsmethods)

This osie override option could be defined [here](https://github.com/tinkerbell/boots/blob/0199cbccfa3ad4f3bdd499e0214c04d31b822660/cmd/boots/main.go#L339) as a CLI flag. This would also automatically make an env var available. The env var would be prefaced with `BOOTS_`. So a full example would be `BOOTS_OSIE_OVERRIDE`.